src/instance-aws.c

```diff
@@ -0,0 +1,208 @@
+/*
```

Comment:
The tuner lib does not link against all the source files currently. It links against `nccl_ofi_regions.c` and `nccl_ofi_tuner.c` . When I tried to link the tuner with platform-aws.c, I had to link almost all the source files from the plugin to the tuner because of the dependencies of platform-aws.c
To avoid this, I moved `get_platform_type` to a new file instance-aws.c.

Reply:
no, this is not the right change then.  Do not just split up code like this.  If we need to solve how to share code between the tuner and plugin, then let's do that.  But this is not the way to do that.